boat-scaffold/src/main/java/com/backbase/oss/codegen/java/BoatSpringCodeGen.java

```diff
@@ -391,4 +393,19 @@ public String toDefaultValue(CodegenProperty cp, Schema schema) {
             .map(CodegenValueType::getValue)
             .orElseGet(() -> super.toDefaultValue(cp, referencedSchema));
     }
+
+    @Override
+    public void postProcessModelProperty(CodegenModel model, CodegenProperty property) {
+        super.postProcessModelProperty(model, property);
+
+        if ("string".equalsIgnoreCase(property.openApiType) && ("number".equals(property.dataFormat) || "integer".equals(property.dataFormat))) {
+            boolean isString = Stream.of(property.baseType, property.dataType, property.datatypeWithEnum)
+                .anyMatch("string"::equalsIgnoreCase);
+            if (!isString && !property.vendorExtensions.containsKey("x-extra-annotation")) {
+                property.vendorExtensions.put("x-extra-annotation", "@JsonSerialize(using = ToStringSerializer.class)");
```

Comment:
For me, using the ToStringSerializer did not work. When the property is a BigDecimal, it will delegate to the NumberSerializer, which will only use 'toPlainString' when the global BigDecimalAsString option is set.

Reply:
ToStringSerializer calls `String.valueOf(value)` which will use the `BigDecimal.toString()` method which in some cases will use scientific notation.